spec/services/rapid_ready_for_decision/hypertension_pdf_generator_spec.rb

```diff
@@ -33,10 +33,18 @@
     RapidReadyForDecision::HypertensionPdfGenerator.new(patient_name, parsed_bp_data, parsed_medications_data)
   end
 
+  let(:pdf_generator_no_meds) do
+    RapidReadyForDecision::HypertensionPdfGenerator.new(patient_name, parsed_bp_data, [])
+  end
+
```

Comment:
@nanotone i doubt this is the best way to do this but couldn't infer from the way it was setup

Reply:
Okay yeah, I see how this is set up. My suggestion is to add a new

```ruby
context 'when no medications are present'
```

block _inside_ the `describe '#generate'` block, and within that context do the override

```ruby
let(:parsed_medications_data) { [] }
```

That way you can reuse the existing `pdf` variable within your two new `it` blocks.

(I'm not that familiar with VCR magic yet, but I think it should work although the cassette might need a little rejiggering.)

Reply:
Also, in the spirit of leaving things nicer than they were, I don't think the `pdf` variable needs to be `let!`. It's referenced in every `it` block, and should really just be a `subject { }`.

Reply:
Nice! Thanks, much cleaner. I think VCR functions at the HTTP request step so should be okay but will regenerate just to make sure nothing weird is going on.

Reply:
Works beautifully. I'm guessing `context` here re-runs 

> Also, in the spirit of leaving things nicer than they were, I don't think the `pdf` variable needs to be `let!`. It's referenced in every `it` block, and should really just be a `subject { }`.

Just saw this, will clean that up.